dagfactory/dagfactory.py

```diff
@@ -106,9 +111,20 @@ def build_dags(self) -> Dict[str, DAG]:
                 dags[dag["dag_id"]]: DAG = dag["dag"]
             except Exception as err:
                 raise DagFactoryException(f"Failed to generate dag {dag_name}. verify config is correct") from err
+            else:
+                self.taskgroups_count = dag_builder.taskgroups_count
```

Comment:
I am wondering what would the count represent here. Is it the count of all the taskgroups across all the DAGs in the deployment? If so, do we need to keep incrementing this count for each of the DAG?

Reply:
Good catch, @pankajkoti , please, take a look at ad1063af6fc97a81d4dd8799ac51dc549a0dc9c8 - I think I addressed the issue and also added a test case to cover the scenario of a YAML file that would contain multiple DAGs.

Reply:
Thanks Tatiana. Looks great and my questions are answered